src/rgw/rgw_rados.cc

```diff
@@ -11884,23 +11884,17 @@ int RGWRados::convert_old_bucket_info(RGWObjectCtx& obj_ctx,
   return 0;
 }
 
-int RGWRados::get_bucket_info(RGWObjectCtx& obj_ctx,
-                              const string& tenant, const string& bucket_name, RGWBucketInfo& info,
-                              real_time *pmtime, map<string, bufferlist> *pattrs)
+int RGWRados::bucket_info_cache_miss(RGWObjectCtx& obj_ctx,
+                                     const string& tenant,
+                                     const string& bucket_name,
+                                     RGWBucketInfo& info,
+                                     real_time *pmtime,
+                                     map<string, bufferlist> *pattrs)
 {
   bucket_info_entry e;
   string bucket_entry;
   rgw_make_bucket_entry_name(tenant, bucket_name, bucket_entry);
```

Comment:
@adamemerson should pass bucket_entry as param, otherwise it's going to be calculated twice on cache miss.